### PR TITLE
world history course page

### PR DESCRIPTION
**Pull request by emilia-friedberg**
## WHAT
Create the World History (1200 to Present) Course page, the various routing structures to get there, and the backend functionality necessary to get the data for this page.

## WHY
This is how teachers will assign and view top-level data about work for our new Social Studies course!

## HOW
Lots of React, some Rails. I think most of what's here is fairly straightforward but effectively the same page appears in three different places, so that took some work to set up.

Note: I may try to generalize the container component, and maybe even the erb files, down the line so that we can reuse them for other courses (there is one more coming soon and one sometime this fall), but I didn't want to over-optimize at this stage and this PR is already pretty hefty.

### Screenshots
Screenshot 2024-07-26 at 11 33 01 AM

### Notion Card Links
https://www.notion.so/quill/World-History-1200-CE-Present-Course-Page-76b5b77730ea48f7ab42cc32a11e680e?pvs=4

### What have you done to QA this feature?
Lots of testing of each page in various states (nothing assigned, some assigned but not completed, some completed, etc), plus a check for mobile compatibility. Jack and Jamie are QAing.

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  YES
Have you deployed to Staging? | YES
Self-Review: Have you done an initial self-review of the code below on Github? | YES

### CODE REVIEW

**Review by happythenewsad — APPROVED** [review]
Nice work!

File: services/QuillLMS/app/helpers/navigation_helper.rb [hunk 1]
```diff
@@ -336,11 +343,17 @@ def mobile_subnav_tabs(current_user)
...
         standards_tab,
-        data_export_tab
-      ]
+        data_export_tab,
+        social_studies_dashboard_tab(current_user)
+      ].compact
     end
   end
 
+  def should_show_social_studies_dashboard_tab?(current_user)
```
Comment (brendanshean) [inline services/QuillLMS/app/helpers/navigation_helper.rb]: `should_` here feels redundant given the boolean suffix `?`.  I would get rid of it.

File: services/QuillLMS/app/models/concerns/teacher.rb [hunk 2]
```diff
@@ -80,6 +80,11 @@ def classroom_ids_i_coteach_or_have_a_pending_invitation_to_coteach
     Set.new.tap { |ids| all_ids.each { |row| ids.merge(row.values) } }
   end
 
+  def unit_activities_for_classrooms_i_teach
```
Comment (brendanshean) [inline services/QuillLMS/app/models/concerns/teacher.rb:83]: I think you can use ActiveRecord to construct the array:

```
def unit_activities_for_classrooms_i_teach
  UnitActivity
    .joins(:classroom_units)
    .where(classroom_units: { classroom_id: classrooms_i_teach.pluck(:id) })
end
```

File: services/QuillLMS/app/views/teachers/shared/_scorebook_tabs.html.erb [hunk 3]
```diff
@@ -97,6 +97,11 @@
             <li class="premium-reports-tab"><%= active_link_to raw('<span>Concepts</span><div class="small-diamond-icon"></div>'), teachers_progress_reports_concepts_students_path%></li>
             <li class="premium-reports-tab"><%= active_link_to raw('<span>Standards</span><div class="small-diamond-icon"></div>'), teachers_progress_reports_standards_classrooms_path %></li>
             <li class="premium-reports-tab"><%= active_link_to raw('<span>Data Export</span><div class="small-diamond-icon"></div>'), teachers_progress_reports_activity_sessions_path %></li>
+
+            <% if should_show_social_studies_dashboard_tab?(current_user) %>
```
Comment (brendanshean) [inline services/QuillLMS/app/views/teachers/shared/_scorebook_tabs.html.erb]: I think the `should_` prefix here is redundant given the boolean suffix.  I would drop it from the method name.

**Review by brendanshean — APPROVED** [review]
This looks awesome.  Nice work on this!

**Review by eadams17 — APPROVED** [review]
This looks so good! Thanks for putting all of this together

**Pull request merged**